marketplace/deployer_util/expand_config.py

```diff
@@ -73,6 +77,10 @@ def expand(values_dict, schema):
       result[k] = generate_password(prop.password)
       continue
 
+    if v is None and prop.xtype == 'APPLICATION_UID' and app_uid:
```

Comment:
move `app_uid` check to an inner `if`, and should raise if it's not specified

Reply:
we can't actually do that, as expand_config.py needs to run before app_uid is specified.

i'll leave a comment explaining why.

Reply:
... or maybe a bogus value is okay...? feels icky.

Reply:
Nevermind - `print_config.py` does not rely on `expand_config.py` having been run. Removed two-pass approach.